juju/application.py

```diff
@@ -241,30 +243,35 @@ def collect_metrics(self):
         log.debug(
             'Getting config for %s', self.name)
 
-        return (await app_facade.Get(self.name)).config
+        return (await app_facade.Get(application=self.name, branch='master')).config
 
     async def get_trusted(self):
         """Return the trusted configuration setting for this application.
 
         """
+        if self.model.info.agent_version < client.Number.from_json('2.4.0'):
+            raise NotImplementedError("trusted is not supported on model version {}".format(self.model.info.agent_version))
+
         app_facade = client.ApplicationFacade.from_connection(self.connection)
 
         log.debug(
             'Getting config for %s', self.name)
 
-        config = await app_facade.Get(self.name)
+        config = await app_facade.Get(application=self.name, branch='master')
         if 'trust' in config.config:
             return config.config['trust']['value'] is True
-        if 'application-config' in config.unknown_fields:
-            app_config = config.unknown_fields['application-config']
-            return app_config['trust']['value'] is True
-        return False
+
+        app_config = config.application_config
+        return app_config['trust']['value'] is True
 
     async def set_trusted(self, trust):
         """Set the trusted configuration of the application.
 
         :param bool trust: Trust the application or not
         """
+        if self.model.info.agent_version < client.Number.from_json('2.4.0'):
```

Comment:
Shouldn't this be done with Facade versions? Is agent version really the only way we can detect this?